spec/acceptance/traits_spec.rb

```diff
@@ -725,3 +725,36 @@ def initialize(name)
     expect(creator.name).to eq 'Joe Creator'
   end
 end
+
+describe "associations set in traits" do
+  before do
+    define_model("Person", name: :string)
+
+    define_model("Book", name: :string, author_id: :integer) do
+      belongs_to :author, class_name: "Person"
+    end
+
+    FactoryGirl.define do
+      factory :person do
+        name "Ploni"
+      end
+
+      factory :bob, parent: :person do
+        name "Bob"
+      end
+
+      factory :book do
+        association :author, factory: :person
+      end
+
+      trait :bob do
```

Comment:
I stumbled over this pull request and was wondering whether a trait can be defined outside of a factory.